src/idx_html.pre.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-disable import/no-unresolved */
+const htmlPre = require('./html.pre');
+
+/**
+ * The 'pre' function that is executed before the HTML is rendered
+ * @param context The current context of processing pipeline
+ * @param context.content The content
+ */
+function pre(context) {
+  const { document } = context.content;
+  htmlPre.pre(context);
+
+  // construct the tables
+  const tables = [];
+  const basic = { name: 'basic', entries: {} };
+  const images = { name: 'images', entries: {} };
+
+  const titleEl = document.querySelector('h1');
+  if (titleEl) {
+    basic.entries.title = titleEl.textContent;
+  }
+
+  const descEl = document.querySelector('.title .header p');
+  if (descEl) {
+    basic.entries.description = descEl.textContent;
+  }
+
+  const imgs = [];
+  document.querySelectorAll('img').forEach((img) => {
+    imgs.push(img.src);
+  });
+  images.entries = { images: imgs };
+  tables.push(basic);
+  tables.push(images);
+  context.content.json = { tables };
+
+  context.content.json.string = JSON.stringify(context.content.json);
+}
+
+module.exports.pre = pre;
+/**
+ * Override fetch step
+ */
+module.exports.before = {
+  fetch: (context, action) => {
+    action.secrets = action.secrets || {};
+    action.secrets.HTTP_TIMEOUT = 5000;
```

Comment:
Why do you need this ?

Reply:
1000 default time is not enough and times out

Reply:
Is this a local issue or does this happen in production, too? The low timeout is a resilience feature because it is preferable to have the renderer fail for a single request quickly (after 1000 ms) rather than block four more concurrent execution slots while waiting for a longer timeout.

If it's a local issue, I'd rather look for a local (testing-only) fix. If it happens in production, we need to investigate.

Reply:
i haven't tried it in prod yet but locally it is an issue for now - let me investigate more

Reply:
ok if we use `utils/preFetch` there is a 5% chance on average we will get `[hlx] error: Gateway timout of 1000 milliseconds exceeded for {url...}` i will test the usage of default preFetch after this PR gets merged in

P.S. timeout is misspelled in the error message

Reply:
@craeyu the local issue should go away when you use PollyJS, which records the request and replays it – the main purpose is to allow offline testing, but making requests faster (by faking them) is a nice side effect.